### DISCUSSION TITLE
What is a InitializeClientSession request in version 2.0.0 ?

### DISCUSSION THREAD

**Discussion by rsoika** [eclipse-glsp/glsp#1150] [Q&A] [answered]
Hi,
migrating from 1.1.0 to 2.0.0 I stumbled over a problem with the old java method `configureClientActions` in my DiagramModule. I used this method to bind my custom action which is send between client and server to inform about model changes:

```java
    @Override
    protected void configureClientActions(MultiBinding<Action> binding) {
        super.configureClientActions(binding);
        binding.add(BPMNPropertyPanelUpdateAction.class);
    }
```

The release notes state: _Remove configureClientActions from DiagramModule as client actions are now implicitly configured via InitializeClientSession request_

Can someone give me a short example how my code should look like in version 2.0.0 ?

**Comment by tortmayr** [comment]
Hi @rsoika,
we a currently still in the progress of updating the documentation for GLSP 2.0 and will also provide a `Migration guide` that will
cover the necessary adaptions from 1.0.0 to 2.0.0 in detail.
I would recommend to wait with the upgrade to 2.0.0 until this migration guide is available.

**Reply by rsoika** [reply]
Ok  no hurry ;-)

**Reply by rsoika** [reply]
Hi @tortmayr, I know I'm a little impatient. But does [this change](https://github.com/eclipse-glsp/glsp-server/commit/9e41d4a2fa5913f0499bbefab795589e0f4beb76) simply mean, that I no longer need to bind my custom action in the method `configureClientActions` on the server side, and it simply is sufficient that my client module configures the corresponding action handler and will now automatically react when my server sends this kind of action?


```javascript
export const BPMNPropertyModule = new ContainerModule((bind, _unbind, isBound, rebind) => {
    bind(BPMNPropertyPanel).toSelf().inSingletonScope();
    bind(TYPES.IUIExtension).toService(BPMNPropertyPanel);

    configureActionHandler({ bind, isBound }, BPMNPropertyPanelUpdateAction.KIND, BPMNPropertyPanel);
});
```

**Reply by tortmayr** [reply, marked as answer]
Hi @rsoika,
yes that is correct. The client now sends a list of all handled action kinds to the server in the `initializeClientSession` Request.
This means it is no longer necessary to explicitly configure Client Actions on the server side. You can simply dispatch any action on the server that has a registered client handler (or command) and it will be automatically forwarded to the client.
